app/notification.go

```diff
@@ -281,6 +283,7 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 
 				membershipsMutex.Lock()
 				participantMemberships[userID] = threadMembership
+				newParticipants[userID] = newMembership
 				membershipsMutex.Unlock()
```

Comment:
can I do that?

Reply:
Do you mean "reusing" the same mutex lock? Yep!